File: examples/java/customers.java
import twila.parcelemais.ParceleMaisClient;
import twila.parcelemais.PagedResult;
import twila.parcelemais.customers.model.Customer;
import twila.parcelemais.customers.model.ListCustomersRequest;

import java.util.UUID;

public final class CustomersExample {

    public static Customer getCustomer(ParceleMaisClient client, UUID customerId) {
        return client.customers().get(customerId);
    }

    public static PagedResult<Customer> findByDocument(ParceleMaisClient client, String cpf) {
        return client.customers().list(ListCustomersRequest.builder()
                .document(cpf)
                .page(1)
                .pageSize(10)
                .build());
    }

    public static void printCustomers(ParceleMaisClient client) {
        PagedResult<Customer> page = findByDocument(client, "12345678900");

        for (Customer customer : page.getItems()) {
            System.out.println(customer.getName() + " - " + customer.getDocument());
        }
    }
}

File: examples/node/customers.ts
import { ParceleMaisClient, ParceleMaisEnvironment } from '@twila/parcelemais';

const client = new ParceleMaisClient({
  clientId: process.env.PARCELEMAIS_CLIENT_ID!,
  clientSecret: process.env.PARCELEMAIS_CLIENT_SECRET!,
  environment: ParceleMaisEnvironment.Staging,
});

export async function findCustomerByDocument(document: string) {
  const page = await client.customers.list({ document, pageSize: 1 });
  return page.items[0];
}

export async function getCustomer(customerId: string) {
  return client.customers.get(customerId);
}
